### ISSUE TITLE
Update Manga element to add Webtoon

### ISSUE THREAD

**Issue by Belphemur** [anansi-project/comicinfo#40]
### Where does this comes from?

Hello,
Currently, the Manga element has the value `Yes` and `YesAndRightToLeft`.

I suggest adding a new one: `YesAndWebtoon` or to rethink the Manga element to have an attribute about the reading "direction".

### What is the rationale for adding support for this element?

Help any kind of reader what should be the default way to read the manga.

**Comment by ajslater** [comment]
IMHO an element specifically for ReadingDirection with four states would be best for this and would inform software how to present the comic. `LeftToRight`, `RightToLeft`, `TopToBottom`, `BottomToTop`. The Manga tag is partially overloaded for part of this already, but the Manga: "No" or "Yes" states by themselves doesn't really tell the presentation software anything useful.
An alternative might be two boolean elements. ReadingOrientation and ReadingDirection.

**Comment by Belphemur** [comment]
I'm for ReadingDirection that makes total sense and in this case we can deprecate the YesAndRighToLeft for manga.

Sent from Nine
________________________________

**Comment by gotson** [comment]
Given this is an Enum value, adding new values could be a breaking change, which we don't want to introduce.

**Comment by Belphemur** [comment]
To be honest I prefer the solution of @ajslater.
I'll rephrase my request.

Adding a new object for reading direction make more sense.

**Comment by gotson** [comment]
> IMHO an element specifically for ReadingDirection with four states would be best for informing software how to present the comic. `LeftToRight`, `RightToLeft`, `TopToBottom`, `BottomToTop`.

The limitation with that is that it doesn't define how the pages should flow. For instance you can't differentiate top to bottom page by page, and top to bottom continuous (webtoon).

I agree that using a new field would be beneficial, since the current `Manga` field is botched.

**Comment by Belphemur** [comment]
I'm curious if there are actual scenarios for non-continious top-down.

Would it always be continuous?

Sent from Nine
________________________________

**Comment by gotson** [comment]
Assuming that all top to bottom publications need to be read in a continuous way would limit the data model. There might also be some publications in LTR or RTL direction that are made to be read in a continuous manner.

In the end the consuming application will try its best to map the metadata to its actual capabilities.

**Comment by Belphemur** [comment]
That make sense.
Then in that case having a boolean attribute saying if it should be continuous would work for the reading direction.

Sent from Nine
________________________________

**Comment by Belphemur** [comment]
I've updated my request with your suggestions @ajslater & @gotson

**Comment by ajslater** [comment]
I've just noticed that the OPDS-2.0 spec's metadata schema has something similar to my proposal:

https://github.com/readium/webpub-manifest/blob/master/schema/metadata.schema.json#L107

```json
    "readingProgression": {
      "type": "string",
      "enum": [
        "rtl",
        "ltr",
        "ttb",
        "btt",
        "auto"
      ],
      "default": "auto"
    },
```

I suppose i like `ReadingDirection` better, but maybe their abbreviated enums have merit? 'auto' seems a bit odd to me. I think i'd consider that case the same as the attribute being omitted.

**Comment by gotson** [comment]
i would not cater for whether it's paged or continuous though.

**Comment by gotson** [comment]
I've just bumped into https://readium.org/webpub-manifest/modules/presentation.html which may be of interest. Haven't had time to read in full yet though.

**Comment by ajslater** [comment]
Yeah they farm out 'continuous' to the presentation extension schema. I've not thought deeply about what they consider part of the presentation schema and what they consider core metadata. There's probably a good argument that reading direction ('progression') is a presentation detail.

I've been implementing OPDS 2.0 for Codex recently on a lark, and I have to say I'm impressed. The OPDS 1.x schema shows its age, but the 2.0 draft spec, imho, is very well thought out. I may be slightly biased because it's pretty close to the Codex API, it seems to be a case of convergent evolution. I feel like I could replace the Codex API with OPDS 2.0 with only minor extensions. 

OPDS 2.0 is missing a page by page API, like OPDS 1.2 has, but explicitly has facility for templated links in their search specification (which is not OpenSearch):

```json
{
  "rel": "search",
  "href": "search{?query}", 
  "type": "application/opds+json", 
  "templated": true
}
```

They also have a specific “image” type link section, but using that for pages feels too comic specific for this publication schema.

A templated schema for pages would be fine. But also, I don’t think book pagination is as open ended as search and therefore might not require a template. 
A specific ‘give me a page’ schema with optional legal min and max values and the likely mime-type of the returned page might even be better.

**Comment by ajslater** [comment]
Woops. The latter half of that post is off topic. I'm actually in a separate discussion with gotson talking about comic server page streaming, OPDS 2.0 and  synchronization.

**Comment by t0815** [comment]
@Belphemur im currently developing a cbz archive editor. There its possible to define your own fields which should be included in metadata. In addition you can specify how the value is handled.  Meaning, you can set possible values for a certain key. This way, you are totally flexible regarding the potential data and values within the comicinfo.xml

**Comment by SecrestTheHuman** [comment]
I'm gonna bump this, because it feels like a particularly important issue. Partially because no comic reader currently seems to adhere to the "YesAndRightToLeft" thing anyways and I think part of that is it's just weird. The accepted standard of "Manga" "YesAndRightToLeft" is ludicrous and makes no sense. Why is there even a manga section if "Yes" doesn't indicate right to left? What other possible use could that section serve? I honestly think it should be deprecated in favor of a "reading direction" setting and maybe a "country of origin" option if people want to sort their comics based on whether or not they're Japanese in origin.

ReadingDirection could additionally be used to specify if a comic could be webtoon format, ie continuous scrolling up and down.

It could take the options "LeftToRight" "RightToLeft" "UpToDown" "DownToUp" "VerticalScroll" or "HorizontalScroll."

To the best of my knowledge some of those aren't relevant to any currently available comics, but I don't want to limit what people can do with digital comics, just give them the best tools to do them.
